### ISSUE TITLE
subMatch doesn't work on AWS Tags with custom checks

### ISSUE THREAD

**Issue by grantkl** [tfsec/tfsec#774]
For some reason the subMatch functionality doesn't work on AWS Tags. 

TFSec Custom Check:
```
---
checks:
  - code: TAG_custom
    description: Custom check to ensure the Environment tag is applied
    errorMessage: The required Environment tag was missing
    matchSpec:
      action: contains
      name: tags
      value: Environment
      subMatch:
        action: equals
        name: Environment
        value: production
    requiredTypes:
    - resource
    severity: ERROR
    requiredLabels:
    - aws_lb
```

Test TF File (this one will pass the check even with the value being wrong):
```
resource "aws_lb" "test" {
  name                       = "test"
  internal                   = true
  drop_invalid_header_fields = "true"

  tags = {
    Environment = "test"
  }
}
```

The check only fails if Environment is missing completely like this:
```
resource "aws_lb" "test" {
  name                       = "test"
  internal                   = true
  drop_invalid_header_fields = "true"

  tags = {
  }
}
```

I tried various options of using predicates, but none of them worked and this is the simplest example that should work based on the documentation.

**Comment by grantkl** [comment]
I created a couple of tests here showing the failure https://github.com/grantkl/tfsec/blob/master/internal/app/tfsec/custom/aws_tag_submatch_test.go

**Comment by owenrumney** [comment]
@grantkl  - the reason this isn't working is because the subMatch assumes that it is going to get a block to work with, I'm just looking at a workaround or fix

**Comment by owenrumney** [comment]
@grantkl - this is fixed in PR #775 

Your check config would look like this;

```yaml
---
checks:
  - code: TAG_custom
    description: Custom check to ensure the Environment tag is applied
    errorMessage: The required Environment tag was missing
    matchSpec:
      action: contains
      name: tags
      value: 
        Environment: "production"
    requiredTypes:
    - resource
    severity: ERROR
    requiredLabels:
    - aws_lb
```
Note the value is a keyvalue pair, if you only wanted to ensure that there was an Environment tag, but didn't care about the value you would just use

```yaml
value: Environment

or 

value:
  Environment
```

**Comment by owenrumney** [comment]
@grantkl  - this is now available in v0.40.1

**Comment by grantkl** [comment]
Nice! I'll try this out this week. 

If I wanted multiple valid values would it look like this?
```
      value: 
        Environment: "production"
        Environment: "test"
```
